modules/programs/swaylock.nix

```diff
@@ -0,0 +1,32 @@
+{ config, lib, ... }:
+
+let cfg = config.programs.swaylock;
+in {
+  meta.maintainers = [ lib.hm.maintainers.rcerc ];
+
+  options.programs.swaylock.settings = lib.mkOption {
+    type = with lib.types; attrsOf (oneOf [ bool float int str ]);
+    default = { };
+    description = ''
+      Default arguments to <command>swaylock</command>. An empty set
+      disables configuration generation.
+    '';
+    example = {
+      color = "808080";
+      font-size = 24;
+      indicator-idle-visible = false;
+      indicator-radius = 100;
+      line-color = "ffffff";
+      show-failed-attempts = true;
+    };
+  };
+
+  config.xdg.configFile."swaylock/config" = lib.mkIf (cfg.settings != { }) {
+    text = lib.concatStrings (lib.mapAttrsToList (n: v:
```

Comment:
maybe you could use `generators.toINIWithGlobalSection ` from nixpkgs since it looks like an ini file.

Reply:
I was hoping to use a generator like that, but swaylock only seems to accept boolean options specified as `name` instead of `name=true` (at least on my system).

Reply:
we should have a way to parametrize this in nixpkgs t o avoid having to rewrite this every time.

Reply:
Maybe, although it might be non-standard INI syntax. A custom `mkKeyValue` function could be passed to handle these cases though. Do you suggest that in this case?